### PR TITLE
Faster full node tests

### PR DESCRIPTION
**Pull request by mariano54**
22 -> 12 mins approximately

For optimal speed, we still need to merge:
https://github.com/Chia-Network/chia-blockchain/pull/10978
or 
https://github.com/Chia-Network/chia-blockchain/pull/10847

There is one real code change here, and that is for the full node to loop with fine grained sleep when someone is sending a transaction, instead of the slow 100ms sleep from before. This will also improve UX of sending TXs.

### CODE REVIEW

File: chia/full_node/full_node_api.py [hunk 1]
```diff
@@ -1264,8 +1264,9 @@ async def send_transaction(self, request: wallet_protocol.SendTransaction, *, te
         )
         # Waits for the transaction to go into the mempool, times out after 45 seconds.
         status, error = None, None
-        for i in range(450):
-            await asyncio.sleep(0.1)
+        sleep_time = 0.01
+        for i in range(int(45 / sleep_time)):
+            await asyncio.sleep(sleep_time)
```
Comment (arvidn) [inline chia/full_node/full_node_api.py:1269]: oh. in an ideal world we wouldn't just blindly sleep here, but we would have a queue for responses that we could wait on.

File: tests/core/full_node/test_mempool.py [hunk 2]
```diff
@@ -282,7 +253,7 @@ async def test_basic_mempool_manager(self, bt, two_nodes_mempool, wallet_a, self
             (ConditionOpcode.ASSERT_HEIGHT_RELATIVE, 0, MempoolInclusionStatus.PENDING),
             (ConditionOpcode.ASSERT_HEIGHT_RELATIVE, 1, MempoolInclusionStatus.PENDING),
             # the absolute height and seconds tests require fresh full nodes to
-            # run the test on. The fixture (two_nodes_mempool) creates 3 blocks,
+            # run the test on. The fixture (one_node_one_block) creates 3 blocks,
```
Comment (arvidn) [inline tests/core/full_node/test_mempool.py:256]: the name of this fixture suggests that it only creates one block (compared to 3 block by `two_nodes_mempool`). If that's right, I would expect the heights below need updating as well.

File: tests/core/full_node/test_mempool.py [hunk 3]
```diff
@@ -311,8 +282,10 @@ def test_fun(coin_1: Coin, coin_2: Coin) -> SpendBundle:
             bundle = SpendBundle.aggregate([tx1, tx2])
             return bundle
 
-        full_node_1, _, server_1, _ = two_nodes_mempool
-        blocks, bundle, status, err = await self.condition_tester2(bt, two_nodes_mempool, wallet_a, test_fun)
+        full_node_1, server_1 = one_node_one_block
+        _ = await next_block(full_node_1, wallet_a, bt)
+        _ = await next_block(full_node_1, wallet_a, bt)
```
Comment (arvidn) [inline tests/core/full_node/test_mempool.py:287]: oh,. I see. you just make two extra blocks here to make the numbers the same as before. you would get less code and faster tests by just decrementing the expected block heights by 2 instead

File: tests/core/full_node/test_mempool.py [hunk 4]
```diff
@@ -565,7 +538,12 @@ async def condition_tester(
             farmer_reward_puzzle_hash=reward_ph,
             pool_reward_puzzle_hash=reward_ph,
         )
-        peer = await connect_and_get_peer(server_1, server_2, bt.config["self_hostname"])
+        _, dummy_node_id = await add_dummy_connection(server_1, bt.config["self_hostname"], 100)
+        dummy_peer = None
+        for node_id, wsc in server_1.all_connections.items():
+            if node_id == dummy_node_id:
+                dummy_peer = wsc
+                break
```
Comment (arvidn) [inline tests/core/full_node/test_mempool.py:546]: it looks like `connect_and_get_peer()` was useful. We do this at least twice. Why remove it?

File: tests/core/full_node/test_mempool.py [hunk 5]
```diff
@@ -772,15 +758,18 @@ async def test_correct_my_id(self, bt, two_nodes_mempool, wallet_a):
...
 
     @pytest.mark.asyncio
-    async def test_my_id_garbage(self, bt, two_nodes_mempool, wallet_a):
+    async def test_my_id_garbage(self, bt, one_node_one_block, wallet_a):
 
-        full_node_1, full_node_2, server_1, server_2 = two_nodes_mempool
+        full_node_1, server_1 = one_node_one_block
+
+        _ = await next_block(full_node_1, wallet_a, bt)
+        _ = await next_block(full_node_1, wallet_a, bt)
```
Comment (arvidn) [inline tests/core/full_node/test_mempool.py:766]: you add these extra two blocks in a bunch of places. I *think* we could just change what the test expects instead, but I might be forgetting something

File: tests/core/full_node/test_full_node.py [hunk 6]
```diff
@@ -806,87 +795,90 @@ async def test_new_transaction_and_mempool(self, wallet_nodes, bt, self_hostname
...
-                receiver_puzzlehash = wallet_receiver.get_new_puzzlehash()
-                puzzle_hashes.append(receiver_puzzlehash)
-                output = ConditionWithArgs(ConditionOpcode.CREATE_COIN, [receiver_puzzlehash, int_to_bytes(100000000)])
-
-                conditions_dict[ConditionOpcode.CREATE_COIN].append(output)
-
-            spend_bundle = wallet_a.generate_signed_transaction(
-                100,
-                puzzle_hashes[0],
-                get_future_reward_coins(blocks[1 + i])[0],
```
Comment (arvidn) [inline tests/core/full_node/test_full_node.py:826]: running this test in different blocks isn't relevant I take it?

File: tests/core/full_node/test_mempool.py (hunk 3 above)
Comment (mariano54) [reply tests/core/full_node/test_mempool.py:287]: It wasn't passing though

File: chia/full_node/full_node_api.py (hunk 1 above)
Comment (mariano54) [reply chia/full_node/full_node_api.py:1269]: Yes true, but this is out of scope of this PR>=

File: tests/core/full_node/test_mempool.py (hunk 2 above)
Comment (mariano54) [reply tests/core/full_node/test_mempool.py:256]: It only creates one block, i'll update the comment

File: tests/core/full_node/test_mempool.py (hunk 5 above)
Comment (mariano54) [reply tests/core/full_node/test_mempool.py:766]: The problem is that there are many many tests, so the ones that require the 3 blocks are probably less than 15% here.

File: tests/core/full_node/test_mempool.py (hunk 4 above)
Comment (mariano54) [reply tests/core/full_node/test_mempool.py:546]: connect and get peer is for two nodes to connect to each other. But we don't actually need 2 nodes for this test. Creating unnecessary node processes slows things down due to the setup time. Here a dummy peer is sufficient.

File: tests/core/full_node/test_full_node.py (hunk 6 above)
Comment (mariano54) [reply tests/core/full_node/test_full_node.py:826]: Yeah that is not important for this test. The point of this section is to fill the mempool. I decreased the size of the mempool so now we don't need to many TXs.

File: tests/core/full_node/test_mempool.py (hunk 4 above)
Comment (arvidn) [reply tests/core/full_node/test_mempool.py:546]: I see. `connect_and_get_peer()` did more than this block

Comment (mariano54) [reply tests/core/full_node/test_mempool.py:546]: yeah but the main problem is the fixture which used to get 2 nodes, now it only gets one.

**Review by arvidn — APPROVED** [review]

**Pull request merged**